tests/pact/caps.repl

```diff
@@ -60,7 +60,7 @@
 
   (defpact test-pact-guards (id:string)
     (step (step1 id))
-    (step (step2 (read-msg "id"))))
+    (step (let ((s2 (step2 (read-msg "id")))) "step2")))
```

Comment:
What does this binding do, and how is it extending the test? Is this suggesting that the return type of `test-pact-guards` should be `:string`?

Reply:
`step` has return type `string`, with this change, we enforce that all steps have the same return type. Hence, the change will return the `"step2"` string. This is a fix to a pre-existing test (on caps).

Reply:
Could you please add some comments to the test describing what's going on here and what exactly is being tested by these odd constructions? (Namely, binding to a name that is never used). Everything else looks good!